src/System.Management.Automation/engine/remoting/server/ServerRunspacePoolDriver.cs

```diff
@@ -2098,19 +2103,25 @@ public void Stop()
 
             internal void DoInvoke()
             {
-#if !CORECLR // TODO:CORECLR - WindowsIdentity.Impersonate() is not available. Use WindowsIdentity.RunImplemented to replace it.
-                // Flow impersonation onto thread if needed.
-                WindowsImpersonationContext impersonationContext = null;
-                if ((_identityToImpersonate != null) &&
-                    (_identityToImpersonate.ImpersonationLevel == TokenImpersonationLevel.Impersonation))
-                {
-                    impersonationContext = _identityToImpersonate.Impersonate();
-                }
-#endif
-
                 try
                 {
-                    _results = _wrappedDebugger.ProcessCommand(_command, _output);
+#if !UNIX
+                    if (_identityToImpersonate != null)
+                    {
+                        WindowsIdentity.RunImpersonated(
+                            _identityToImpersonate.AccessToken,
+                            () =>
+                            {
+                                _results = _wrappedDebugger.ProcessCommand(_command, _output);
+                            });
+                    }
+                    else
+                    {
+#endif
+                        _results = _wrappedDebugger.ProcessCommand(_command, _output);
+#if !UNIX
+                    }
+#endif
```

Comment:
How about organize the code this way:
```c#
            internal void DoInvoke()
            {
                try
                {
#if !UNIX
                    if (_identityToImpersonate != null)
                    {
                        WindowsIdentity.RunImpersonated(
                            _identityToImpersonate.AccessToken,
                            () => _results = _wrappedDebugger.ProcessCommand(_command, _output));
                        return;
                    }
#endif
                    _results = _wrappedDebugger.ProcessCommand(_command, _output);
                }
                catch (Exception e)
                {
                    _exception = e;
                }
                finally
                {
                    _commandCompleteEvent.Set();
                }
            }
        }
```

Reply:
Thanks, will do.

Also, I did some experiments and was able to force impersonation of a different account, and this (RunImpersonated) works without error.  I found four other places where we flow impersonation, so I'll make changes in this PR for those locations as well.